### PR TITLE
MGMT-2450 Add email domain label to cluster and host metrics

### PR DESCRIPTION
**Pull request by nmagnezi**

### CODE REVIEW

File: pkg/auth/authz_handler.go [hunk 1]
```diff
@@ -118,6 +120,24 @@ func OrgIDFromContext(ctx context.Context) string {
...
+	return payload.Email
+}
+
+// EmailDomainFromContext returns email Domain from the specified context
+func EmailDomainFromContext(ctx context.Context) string {
+	domain := "Unknown"
+	email := EmailFromContext(ctx)
+	delimiterIdx := strings.LastIndex(email, EmailDelimiter)
+	if delimiterIdx >= 0 {
+		emailElements := strings.Split(email, "@")
```
Comment (ErezAlster) [inline pkg/auth/authz_handler.go:135]: why not to split and then check the size of the array?

Comment (nmagnezi) [reply pkg/auth/authz_handler.go:135]: what cases will it catch that this method won't?

Comment (ErezAlster) [reply pkg/auth/authz_handler.go]: just searching for the string twice is less efficiant...
Nothing too important

Comment (nmagnezi) [reply pkg/auth/authz_handler.go]: multiple `@` signs are allowed in an email address, in which the last `@` should be used as the delimiter between the local part and domain. This is why I used `LastIndex`.

quote:
`
space and special characters "(),:;<>@[\] are allowed with restrictions (they are only allowed inside a quoted string, as described in the paragraph below, and in addition, a backslash or double-quote must be preceded by a backslash);
`

source: https://en.wikipedia.org/wiki/Email_address#Local-part

**Comment by nmagnezi** [comment]
/unhold
Tested Ok, ready for reviews:
Auth enabled:
```
service_assisted_installer_cluster_creations{clusterId="67f5d634-52b8-49ea-8f55-16077c458f03",emailDomain="redhat.com",openshiftVersion="4.6"} 1
service_assisted_installer_cluster_host_cores_bucket{clusterId="67f5d634-52b8-49ea-8f55-16077c458f03",emailDomain="redhat.com",openshiftVersion="4.6",result="Done",role="master",le="4"} 2
service_assisted_installer_cluster_host_cores_count{clusterId="67f5d634-52b8-49ea-8f55-16077c458f03",emailDomain="redhat.com",openshiftVersion="4.6",result="Done",role="master"} 2
service_assisted_installer_cluster_host_cores_sum{clusterId="67f5d634-52b8-49ea-8f55-16077c458f03",emailDomain="redhat.com",openshiftVersion="4.6",result="Done",role="master"} 8
service_assisted_installer_cluster_host_nic_gb_bucket{clusterId="67f5d634-52b8-49ea-8f55-16077c458f03",emailDomain="redhat.com",openshiftVersion="4.6",result="Done",role="master",le="+Inf"} 4
service_assisted_installer_cluster_host_nic_gb_sum{clusterId="67f5d634-52b8-49ea-8f55-16077c458f03",emailDomain="redhat.com",openshiftVersion="4.6",result="Done",role="master"} -4
service_assisted_installer_cluster_host_nic_gb_count{clusterId="67f5d634-52b8-49ea-8f55-16077c458f03",emailDomain="redhat.com",openshiftVersion="4.6",result="Done",role="master"} 4
service_assisted_installer_cluster_host_ram_gb_bucket{clusterId="67f5d634-52b8-49ea-8f55-16077c458f03",emailDomain="redhat.com",openshiftVersion="4.6",result="Done",role="master",le="+Inf"} 2
service_assisted_installer_cluster_host_ram_gb_sum{clusterId="67f5d634-52b8-49ea-8f55-16077c458f03",emailDomain="redhat.com",openshiftVersion="4.6",result="Done",role="master"} 32
service_assisted_installer_cluster_host_ram_gb_count{clusterId="67f5d634-52b8-49ea-8f55-16077c458f03",emailDomain="redhat.com",openshiftVersion="4.6",result="Done",role="master"} 2
service_assisted_installer_cluster_hosts{clusterId="67f5d634-52b8-49ea-8f55-16077c458f03",diskType="HDD",emailDomain="redhat.com",openshiftVersion="4.6",product="KVM",result="Done",role="master",vendor="Red Hat"} 2
service_assisted_installer_cluster_installation_seconds_bucket{clusterId="67f5d634-52b8-49ea-8f55-16077c458f03",emailDomain="redhat.com",openshiftVersion="4.6",result="installed",le="+Inf"} 1
service_assisted_installer_cluster_installation_seconds_sum{clusterId="67f5d634-52b8-49ea-8f55-16077c458f03",emailDomain="redhat.com",openshiftVersion="4.6",result="installed"} 1305.506544605
service_assisted_installer_cluster_installation_seconds_count{clusterId="67f5d634-52b8-49ea-8f55-16077c458f03",emailDomain="redhat.com",openshiftVersion="4.6",result="installed"} 1
service_assisted_installer_cluster_installation_started{clusterId="67f5d634-52b8-49ea-8f55-16077c458f03",emailDomain="redhat.com",openshiftVersion="4.6"} 1
service_assisted_installer_host_installation_phase_seconds_bucket{clusterId="67f5d634-52b8-49ea-8f55-16077c458f03",discoveryAgentVersion="latest",diskType="HDD",emailDomain="redhat.com",openshiftVersion="4.6",phase="Configuring",product="KVM",result="Done",vendor="Red Hat",le="+Inf"} 1
service_assisted_installer_host_installation_phase_seconds_sum{clusterId="67f5d634-52b8-49ea-8f55-16077c458f03",discoveryAgentVersion="latest",diskType="HDD",emailDomain="redhat.com",openshiftVersion="4.6",phase="Configuring",product="KVM",result="Done",vendor="Red Hat"} 168.924255955
service_assisted_installer_host_installation_phase_seconds_count{clusterId="67f5d634-52b8-49ea-8f55-16077c458f03",discoveryAgentVersion="latest",diskType="HDD",emailDomain="redhat.com",openshiftVersion="4.6",phase="Configuring",product="KVM",result="Done",vendor="Red Hat"} 1
```

Auth disabled:
```
service_assisted_installer_cluster_installation_started{clusterId="98d15298-f296-4ce0-b767-ecc0d56957f2",emailDomain="Unknown",openshiftVersion="4.6"} 1
```

**Comment by nmagnezi** [comment]
/hold

retesting host metrics.
will report results soon and remove the hold label.

**Comment by nmagnezi** [comment]
/unhold

Tested okay. all metrics (both cluster and host) have an email domain label.

File: pkg/auth/authz_handler.go [hunk 2]
```diff
@@ -118,6 +123,24 @@ func OrgIDFromContext(ctx context.Context) string {
 	return payload.Organization
 }
 
+// EmailFromContext returns email from the specified context
+func EmailFromContext(ctx context.Context) string {
```
Comment (gamli75) [inline pkg/auth/authz_handler.go:127]: do we have a test for this? maybe need to check an empty string and valid string

Comment (nmagnezi) [reply pkg/auth/authz_handler.go:127]: Done

**Comment by nmagnezi** [comment]
/test images

The images failure has nothing to do with this PR specifically:
`
could not run steps: step src failed: could not wait for build: the build src failed after 54s with reason DockerBuildFailed: Dockerfile build strategy has failed`

**Comment by nmagnezi** [comment]
Rebased

**Pull request merged**